### DISCUSSION TITLE
Update scalable quantity

### DISCUSSION THREAD

**Discussion by vasilevrv** [bigbridge-nl/product-import#38] [Q&A]
Hello!

We use your module, everything works well, except Scalable Quantity updating. Tell me how to do it correctly?

Thanks in advance!

For update qty I use the next code:

```
$product->defaultStockItem()->setQty($quantity);
$product->defaultStockItem()->setIsInStock($stock);
$product->defaultStockItem()->setMinimumQuantity(1);
```

**Comment by patrick-bigbridge** [comment]
Hi @vasilevrv,

I think you mean "salable quantity", and this issue may help you: https://github.com/bigbridge-nl/product-import/issues/36

It mainly involves reindexing.

**Reply by vasilevrv** [reply]
Thanks for your reply! I tried reindexing, even manually, but it doesn't help, although initially it seems to work. Very strange, I will think...

**Reply by vasilevrv** [reply]
Hm, most likely, need to set availability not only for defaultStockItem(), but also for this sourceItem("default"), together. Until I figured out the difference, but it seems to work...

**Reply by garfix** [reply]
Good question. Well `defaultStockItem()` is the pre-MSI Magento stock, while `sourceItem("default")` is the default MSI stock item. But I am not sure if you still need `defaultStockItem()` if you're using MSI.
